sherpa/astro/tests/test_astro_data2.py

```diff
@@ -816,3 +816,22 @@ def test_pha_invalid_units(invalid, make_test_pha):
         pha.units = invalid
 
     assert str(de.value) == f"unknown quantity: '{invalid}'"
+
+
+@requires_fits
+@requires_data
+def test_xmmrgs_notice(make_data_path):
+    '''Test that notice and ignore works on XMMRGS dataset, which is
+    ordered in increasing wavelength, not energy'''
+    from sherpa.astro.io import read_pha, read_rmf
+    dat = read_pha(make_data_path('xmmrgs/P0112880201R1S004SRSPEC1003.FTZ'))
+    rmf = read_rmf(make_data_path('xmmrgs/P0112880201R1S004RSPMAT1003.FTZ'))
+    dat.set_rmf(rmf)
+    dat.units = 'wave'
+    dat.notice(18.8, 19.2)
+    assert len(dat.get_dep(filter=True)) == 41
+    assert dat.get_filter(format='%.2f') == '19.20:18.80'
```

Comment:
We want get_filter to return 18.80:19.20 here, which is a problem.

My bet is that the old check in `get_filter` is no-longer needed with this dataset

https://github.com/sherpa/sherpa/blob/a6f8cc20091d9675cfb5f45054e0c11850d01926/sherpa/astro/data.py#L3428-L3432

I bet we still need it for other datasets (e.g. Chandra) since if I add `raise NotImplementedError` within the  if check then a number of tests fail (and they also fail if I comment out the section, which is more useful), so we need to change the

    if self.units == 'wavelength':

to something like the untested

    if x[-1] < x[0]: